ironfist/xml/map_xml.xsd

```diff
@@ -80,17 +79,44 @@
       <xs:element name="field_E8" type="xs:int"/>
     </xs:sequence>
   </xs:complexType>
-  
+
+  <xs:complexType name="tableElement_t">
+    <xs:attribute name="key" type="xs:string"/>
+    <xs:attribute name="value" type="xs:string"/>
+    <xs:attribute name="type" type="xs:string" default="string"/>
+  </xs:complexType>
+
+  <xs:complexType name="table_t">
+    <xs:sequence>
+      <xs:element name="tableId" type="xs:ID"/>
+      <xs:element name="tableElement" type="tableElement_t" minOccurs="0" maxOccurs="unbounded"/>
+      <xs:element name="parentTable" type="xs:IDREF" minOccurs="0" maxOccurs="1"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="mapVariable_t">
+    <xs:sequence>
+      <xs:element name="table" type="table_t" minOccurs="0" maxOccurs="unbounded"/>

```

Comment:
The way it's stored should mirror the actual meaning unless there's a pressing reason not to. A mapVar is defined as being either a table or a primitive type, so I'd expect to see a use of xs:choice in here.